vuepress/docs/next/tutorials/create/mfe/widget-configuration.md

```diff
@@ -16,251 +17,260 @@ Start by adding a configuration option to an existing MFE. If you don't already
-    if (!Object.values(ATTRIBUTES).includes(name)) {
-      throw new Error(`Untracked changed attribute: ${name}`);
+        this.root = createRoot(this.mountPoint);
+        this.render();
     }
-    if (this.mountPoint && newValue !== oldValue) {
-      this.render();
+
+    attributeChangedCallback(attribute, oldValue, newValue) {
+        if (!WidgetElement.observedAttributes.includes(attribute)) {
+            throw new Error(`Untracked changed attributes: ${attribute}`)
+        }
+        if (this.mountPoint && newValue !== oldValue) {
+            this.render();
+        }
     }
-  }
 
-  connectedCallback() {
-    this.mountPoint = document.createElement('div');
-    this.appendChild(this.mountPoint);
-    this.render();
-  }
-    
-  render() {
-    const name = this.getAttribute(ATTRIBUTES.name);
-    ReactDOM.render(<App name={name} />, this.mountPoint);
-  }
-}
+    render() {
+        const attributeConfig = this.getAttribute(ATTRIBUTES.config);
+        const config = attributeConfig && JSON.parse(attributeConfig);
 
-customElements.define('my-widget', WidgetElement);
+        this.root.render(
+            <App config={config} />
+        );
+    }
+}
 
-export default WidgetElement;
+customElements.define('simple-mfe', WidgetElement);
 ```
 
-2. Replace the contents of `src/App.js` with the following. This component now displays the `name` property, turning the static component from the [React MFE tutorial](./react.md) into a more dynamic component.
+2. Replace the contents of `src/App.js` with the following. This component now receives the `config` property and displays the `name` parameter from it. This turns the static component from the [React MFE tutorial](./react.md) into a more dynamic component. 
 
 ``` javascript
-    import React from 'react';
-    import './App.css';
-
-    function App({name}) {
-      return (
-        <div className="App">
-          <header className="App-header">
-            <p>
-              Hello, {name}!
-            </p>
-          </header>
-        </div>
-      );
-    }
+import './App.css';
+
+function App({config}) {
+  const { params } = config || {};
+  const { name } = params || {};
+
+  return (
+      <div className="App">
+        <header className="App-header">
+          <p>
+            Hello, {name}!
+          </p>
+        </header>
+      </div>
+  );
+}
 
-    export default App;
+export default App;
 ```
 
-3. For test purposes, replace the contents of `public/index.html` with the following. This allows you to set a value for the `name` attribute of the custom element.
+3. For test purposes, add a configuration file `public/mfe-config.json` with the following content:
+``` javascript
+{
+    "params": {
+        "name": "Jane Smith"
+    }
+}
+```
 
+4. Replace the `body` of `public/index.html` with the following. This allows you to set the MFE `config` attribute and test locally with the same configuration structure provided by Entando.
```

Comment:
and test locally with the same configuration structure provided by Entando. >> and locally test the same configuration structure provided by Entando. ?

Reply:
@jyunmitch I don't think that has quite the same meaning. The idea is to test the MFE locally in a way that is similar to what Entando does. We're not actually testing the config structure, we're just setting up the MFE locally in the same way so we can test the MFE.

Reply:
the distinction based on the placement of the adverb is pretty subtle... what is the difference between "locally test" and "test locally"? the latter "sounds" right, but i'm struggling to put words to why

Reply:
@nshaw

Reply:
@Lyd1aCla1r3 It's more about what is being tested - the MFE or the config structure. We're locally testing the MFE with the Entando structure, rather than locally testing the Entando config structure itself. If we changed Jinah's proposal to "locally test the config MFE with the same structure provided by Entando" then I'd be fine with the new language.

Reply:
And, I agree, it's subtle.